mtp_noms_ops/apps/security/forms/check.py

```diff
@@ -112,20 +110,16 @@ def __init__(self, object_id, request, **kwargs):
     @lru_cache()
     def get_object(self):
         """
-        Gets the security detail object, a sender or prisoner profile
+        Gets the check object
         :return: dict or None if not found
         """
         try:
             obj = self.session.get(self.get_object_endpoint_path()).json()
             convert_dates_obj = convert_date_fields(obj, include_nested=True)
             convert_dates_obj['needs_attention'] = convert_dates_obj['credit']['started_at'] < self.need_attention_date
             return obj
-        except HttpNotFoundError:
-            self.add_error(None, _('Not found'))
-            return None
         except RequestException:
-            self.add_error(None, _('This service is currently unavailable'))
```

Comment:
Would we not want to display an error to the user in this case?

Reply:
we would but we cannot, because you can only use `add_error` after a form has been "cleaned"

Reply:
if this method returns None (some kind of error situation), the raises a 404 error. not identical behaviour (different message seen by user), but the line i've removed actually doesn't work.